### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by tibo-mllr**
# Issue Number:

Fix issue #34 

# Description:

Use redux to create a store in order to share the Query Plan between pages

# How to test:

1. Install the new dependencies
2. (Optional): install the browser extension `Redux DevTools`
3. Run the app and it still works!

# Other notes:

Little refactor of the `lib` folder that was starting to become messy

### CODE REVIEW

**Comment by tibo-mllr** [comment]
Maybe I should put the `StoreProvider` in the `components` folder or with the `redux` part instead of the app?

File: lib/redux/queryPlanreducer.ts
Comment (TimotheeVrg) [inline lib/redux/queryPlanreducer.ts:1]: Why queryPlanreducer.ts and not queryPlanReducer.ts ?

Comment (tibo-mllr) [reply lib/redux/queryPlanreducer.ts:1]: Because I might have forgotten to push on the caps :+1:

**Comment by TimotheeVrg** [comment]
Yes the StoreProvider in a `components` folder should be a good idea

**Review by TimotheeVrg — APPROVED** [review]

**Pull request merged**